### PR TITLE
feat: Cool FAQ and Common Errors

### PR DESCRIPTION
**Pull request by Kitzunu**

### CODE REVIEW

File: docs/FAQ.md [hunk 1]
```diff
@@ -1,48 +1,109 @@
...
-#### 1 - How can I contribute? I am not a developer, I know nothing useful.
+If you have trouble installing or compiling AzerothCore read [Common Errors](common-errors.md).
 
-- You can help us by testing our [Pull Requests](Contribute#how-to-test-a-pull-request), and participating on the github issues' discussions. You can also provide support when you can in our discord, on the popular [stackoverflow](https://stackoverflow.com/questions/tagged/azerothcore) and on other emulation forums (ac-web.org for example). You can also help us with documentation (this wiki, or the readme), we need help with that too!
-- You can put bounties (money) on issues and suggestions by using bountysource. Read this: [Bountysource](Bountysource.md)
-- If you're a developer, check out this page: [Contribute](Contribute.md)
+| Did this FAQ not answer your questions? Read [How to ask for help](how-to-ask-for-help.md) on how to proceed with your question in the best way. |
+| --- |
 
-#### 2 - Why is AzerothCore using more RAM than TrinityCore?
```
Comment (FrancescoBorzi) [inline docs/FAQ.md:9]: was this moved somewhere or just removed? I'd keep it tbh

File: docs/FAQ.md [hunk 2]
```diff
@@ -1,48 +1,109 @@
...
+## General-related FAQ
 
-See [Memory Usage](Memory-Usage.md).
+- What operating systems are supported?
```
Comment (FrancescoBorzi) [inline docs/FAQ.md:10]:
```suggestion
- What operating systems/platforms are supported?
```

as Docker is not an OS

File: docs/FAQ.md [hunk 3]
```diff
@@ -1,48 +1,109 @@
...
+  - You can help fixing issues by submitting Pull Requests, read more [here](contribute.md).
+  - You can help us by testing our [Pull Requests](contribute#how-to-test-a-pull-request) and participating on the [github issues' discussions](https://github.com/azerothcore/azerothcore-wotlk/issues).
+  - You can put bounties (money) on issues or suggestions that you want fixed using bountysource. Learn more here: [Bountysource](bountysource.md)
+  - You can help improve the wiki by submitting a [Pull Request](https://github.com/azerothcore/wiki).
 
-#### 5 - Why is it better to ask on Stackoverflow for support questions?
+- Why do you not merge my fix?
+  - All fixes must be reviewed by a dev. All devs don't know everything, so you need to wait until someone reviews it.
+  - Some fixes requires testing and not all devs can test so you need to wait on someone else to test it.
+  - They get merged eventually after getting the To Be Merged label.
```
Comment (FrancescoBorzi) [inline docs/FAQ.md:31]: maybe worth mentioning https://www.azerothcore.org/wiki/merge-process

File: docs/FAQ.md (hunk 1 above)
Comment (Kitzunu) [reply docs/FAQ.md:9]: #13

File: docs/faq.md [hunk 4]
```diff
@@ -2,53 +2,113 @@
...
+  - All fixes must be reviewed by a dev. All devs don't know everything, so you need to wait until someone reviews it.
+  - Some fixes requires testing and not all devs can test so you need to wait on someone else to test it.
+  - They get merged eventually after getting the To Be Merged label.
+  - Maybe you are not following the [SQL/C++ standards](https://www.azerothcore.org/wiki/standard-operating-procedure).
+  - Read more in [Merge Process](merge-process.md).
 
-#### 5 - Why is it better to ask on Stackoverflow for support questions?
+- How do I report a crash?
+  - By pasting your crashlog into a PasteBin or Gist.
+  - The crashlog **needs to come from a RelWithDebInfo or Debug compilation**. If it is from Release it is useless.
```
Comment (FrancescoBorzi) [inline docs/faq.md:41]: it's worth linking: https://www.azerothcore.org/wiki/how-to-restart-and-debug

File: docs/faq.md [hunk 5]
```diff
@@ -2,53 +2,113 @@
...
 
-- Because your question (and its solutions) **stays there** and you or other users **can find it later**, even using **google**
-- Because your question can easily **get lost in the chat**, and asking in the chat can only get answers by the users who are **currently online**
-- Because it's **super easy** (you can log in Stackoverflow using your **Google** or **Facebook** account - **NO NEED TO CREATE A NEW ACCOUNT**)
-- Because you can add common **tags** to your questions (like `c++`, `sql`, `docker`, `lua`, etc...) and have the chance to  **receive help from users** that you would not find in the AC discord chat
-- Because it gives extra **visibility** to our project
-- Because it helps to **avoid duplicate questions**
-- Because by posting questions or answers you can earn **points** (a profile with many points Stackoverflow can help A LOT when you want to find a job as a developer)
+- Do you support Repacks based on AzerothCore?
+  - No. Compile the code yourself if you want support.
```
Comment (FrancescoBorzi) [inline docs/faq.md:44]:
```suggestion
  - No. Repacks are NOT supported and we strongly suggest to not use them for [several reasons](https://www.mangosrumors.org/why-you-should-not-use-repacks-to-run-your-wow-server/). You can check [this tutorial](https://www.chromiecraft.com/how-to-install-a-wow-server-on-your-own-computer/) for an easy way of installing AC without using any repack.
```

Comment (Kitzunu) [reply docs/faq.md:44]: Me wanting to be a bitch to repacks users :(

File: docs/faq.md [hunk 6]
```diff
@@ -2,53 +2,113 @@
...
+  - The database is updated nearly every day.
 
+- How do I update the database?
+  - You can find everything you need to keep the database up-to-date in this guide on [Database Keeping the Server Up-to-Date](database-keeping-the-server-up-to-date.md).
 
+## Core-related FAQ
 
-## COMMON ERRORS
+- When will the source be stable?
+  - Soon...™
```
Comment (FrancescoBorzi) [inline docs/faq.md:57]:
```suggestion
  - We do our best to keep the `master` branch stable and playable. We never push code directly into the `master` branch, but rather we first require everyone (including AC admins and staff) to first open a PR so everyone is able check them before they got merged into `master`. Please help us by [testing PRs](https://www.azerothcore.org/wiki/How-to-test-a-PR) and reporting any bugs you may find.
 - We recommend updating your core frequently, at least once a week, in order to benefit of latest core fixes and security patches.
```

File: docs/faq.md [hunk 7]
```diff
@@ -2,53 +2,113 @@
...
+  - For OpenSSL versions prior to 1.1.0 they will be named "libeay32.dll" and "ssleay32.dll".
+  - For OpenSSL versions 1.1.0 and more recent they are called:
+    - For 32bit installs: "libssl-1_1.dll" and "libcrypto-1_1.dll".
+    - for 64bit installs: "libssl-1_1-x64.dll" and "libcrypto-1_1-x64.dll"
+  - You can follow the [Core Installation](core-intallation.md) guide to find the libraries.
+
+## Debug-related FAQ
+
+- How can I get a good crashlog on Windows?'
+  - Compile your core in RelWithDebInfo or Debug. A crashlog from Release will be useless.
```
Comment (FrancescoBorzi) [inline docs/faq.md:94]: https://www.azerothcore.org/wiki/how-to-restart-and-debug

File: docs/faq.md (hunk 6 above)
Comment (Kitzunu) [reply docs/faq.md:57]: Fine, but I am not removing Soon...TM, too much of a classic to be forgotten :D

File: docs/faq.md (hunk 7 above)
Comment (Kitzunu) [reply docs/faq.md:94]: Dont think you can do that on windows? Better link https://www.azerothcore.org/wiki/run-worldserver-and-authserver-in-visual-studio

File: docs/faq.md [hunk 8]
```diff
@@ -2,53 +2,113 @@
...
+
+## Module-related FAQ
+
+- I need a new hook for my custom module, what can I do?
+  - You can add the hook to your own fork (cf: [Create a new Hook](create-a-new-hook.md)) and create a new Pull Request to the official repository so we can validate it and merge it.
```
Comment (FrancescoBorzi) [inline docs/faq.md:99]: it's also worth linking this: https://stackoverflow.com/questions/66340549/is-it-possible-to-turn-a-core-patch-into-a-module-for-azerothcore/66340683#66340683

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**